### DISCUSSION TITLE
What is the correct way to query a Role?

### DISCUSSION THREAD

**Discussion by vdkdamian** [netreconlab/Parse-Swift#44] [Q&A] [answered]
What is the correct way to query a `Role`?
I was trying like this below, but I get an `Internal server error` back.
I looked through the Playgrounds but that only show how to save.

```
let roleQuery = Role<User>.query("name" == "myCustomRole")
let role = try await roleQuery.first()
```

**Comment by cbaker6** [comment, marked as answer]
The playgrounds has examples of how to use most of the features: https://github.com/netreconlab/Parse-Swift/blob/cabf542bdd4c8aecf2e162566e57405e9a20ec49/ParseSwift.playground/Pages/12%20-%20Roles%20and%20Relations.xcplaygroundpage/Contents.swift#L173-L193

You can also query like a standard query:

```swift
let allRoleUsers = Role<User>.query(isNotNull(key: "name"))
let result = try await allRoleUsers.first()
```

Be sure your CLP's and ACL's are setup correctly

**Reply by vdkdamian** [reply]
It's indeed a normal query I need. 
I need the Role with a specific name. 
Not the users in that Role.

But keep getting the internal server error when trying to query.

Also think my ACL's are correct. CLP's are set to authenticated only

**Reply by vdkdamian** [reply]
This can be closed. It's definitely an issue on my side.

**Reply by cbaker6** [reply]
It might be helpful to post what your issue was to help others who come across this discussion.

**Reply by vdkdamian** [reply]
> It might be helpful to post what your issue was to help others who come across this discussion.

Well, I actually don't know what was wrong. I just noticed that I was able to find other `Roles` with a specific name.

I was able to solve it by removing that row in the database and recreating it manually.

**Reply by cbaker6** [reply]
That sounds like your ACL was set to be too restrictive

**Reply by vdkdamian** [reply]
Don't really know. I looked at it and it was set like the other roles. But it didn't work.
